amd_openvx/openvx/ago/ago_haf_cpu_arithmetic.cpp

```diff
@@ -3169,7 +3169,8 @@ int HafCpu_AbsDiff_S16_S16S16_Sat
 
 			for (int width = 0; width < postfixWidth; width++)
 			{
-				*pLocalDst++ = (vx_int16)abs((vx_int32)(*pLocalSrc1++) - (vx_int32)(*pLocalSrc2++));
+				vx_int32 temp = (vx_int32) abs((vx_int32) (*pLocalSrc1++) - (vx_int32) (*pLocalSrc2++));
```

Comment:
since we do the abs on s16 inputs, the output never exceeds the S16_min and s16_max. Adding extra check will reduce the performance and hence not needed

Reply:
Yes. You are right. We will switch to your change